From 3a96e09576fd8e6e8a09fb07e4fb5bc27b855787 Mon Sep 17 00:00:00 2001
From: Steve MacLean <sdmaclea.qdt@qualcommdatacenter.com>
Date: Tue, 28 Nov 2017 15:10:38 -0500
Subject: [PATCH] [Arm64] Revise REG_FASTTAILCALL_TARGET

REG_IP0 is used in genEpilogRestore
Use REG_IP1 for FastTailCall target
---
 src/jit/target.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/jit/target.h b/src/jit/target.h
index 1753f44..2f7684d 100644
--- a/src/jit/target.h
+++ b/src/jit/target.h
@@ -1600,8 +1600,8 @@ typedef unsigned short regPairNoSmall; // arm: need 12 bits
   #define RBM_CALLEE_TRASH        (RBM_INT_CALLEE_TRASH | RBM_FLT_CALLEE_TRASH)
   #define RBM_CALLEE_TRASH_NOGC   (RBM_R12|RBM_R13|RBM_R14|RBM_R15|RBM_IP1)
   #define REG_DEFAULT_HELPER_CALL_TARGET REG_R12
-  #define REG_FASTTAILCALL_TARGET REG_IP0   // Target register for fast tail call
-  #define RBM_FASTTAILCALL_TARGET RBM_IP0
+  #define REG_FASTTAILCALL_TARGET REG_IP1   // Target register for fast tail call
+  #define RBM_FASTTAILCALL_TARGET RBM_IP1
 
   #define RBM_ALLINT              (RBM_INT_CALLEE_SAVED | RBM_INT_CALLEE_TRASH)
   #define RBM_ALLFLOAT            (RBM_FLT_CALLEE_SAVED | RBM_FLT_CALLEE_TRASH)
-- 
2.7.4

